openfe/setup/methods/openmm/equil_rbfe_methods.py

```diff
@@ -0,0 +1,838 @@
+            windows=alchem_settings.lambda_windows
+        )
+
+        ## Create the multistate reporter
+        # Get the sub selection of the system to print coords for
+        selection_indices = hybrid_factory.hybrid_topology.select(
+                self._settings.simulation_settings.output_indices
+        )
+
+        # Create the multistate reporter
+        reporter = multistate.MultiStateReporter(
+            self._settings.simulation_settings.output_filename,
+            analysis_particle_indices=selection_indices,
+            checkpoint_interval=self._settings.simulation_settings.checkpoint_interval.m,
+        )
+
+        ## Get platform and context caches
+        platform = _rbfe_utils.compute.get_openmm_platform(
+            self._settings.engine_settings.compute_platform
+        )
+
+        # Create context caches (energy + sampler)
+        # Note: these needs to exist on the compute node
+        energy_context_cache = openmmtools.cache.ContextCache(
+            capacity=None, time_to_live=None, platform=platform,
+        )
+
+        sampler_context_cache = openmmtools.cache.ContextCache(
+            capacity=None, time_to_live=None, platform=platform,
+        )
+
+        ## Set the integrator
+        # get integrator settings
+        integrator_settings = self._settings.integrator_settings
+
+        # create langevin integrator
+        integrator = openmmtools.mcmc.LangevinSplittingDynamicsMove(
+            timestep=integrator_settings.timestep,
+            collision_rate=integrator_settings.collision_rate,
+            n_steps=integrator_settings.n_steps.m,
+            reassign_velocities=integrator_settings.reassign_velocities,
+            n_restart_attempts=integrator_settings.n_restart_attempts,
+            constraint_tolerance=integrator_settings.constraint_tolerance,
+        )
+
+        ## Create sampler
+        # TODO - this needs to variably allow for a SAMS sampler too
+        sampler = _rbfe_utils.multistate.HybridRepexSampler(
+            mcmc_moves=integrator,
+            hybrid_factory=hybrid_factory,
+        )
+
+        # setup sampler and feed it the context caches
+        sampler.setup(
+            reporter=reporter,
+            platform=platform,
+            lambda_protocol=lambdas,
+            temperature=to_openmm(self._settings.integrator_settings.temperature),
+            endstates=alchem_settings.unsampled_endstates,
+        )
+
+        sampler.energy_context_cache = energy_context_cache
+        sampler.sampler_context_cache = sampler_context_cache
+
+        if not dry:
+            # TODO
+            # add some logging for verbosity
+
+            # minimize
+            sampler.minimize(max_iterations=self._settings.simulation_settings.minimization_steps)
+
+            # equilibrate - possibly on a tdqm loop so you can track progress?
+
+            # production - possibly on a tqdm loop so you can track progress?
```

Comment:
```suggestion
            # Get simulation control settings
            sim_settings = self._settings.simulation_settings
            ts = self._settings.integrator_settings.timestep
            mc_steps = self._settings.integrator_settings.n_steps.m

            # minimize
            sampler.minimize(max_iterations=sim_settings.minimization_steps)

            # equilibrate
            equil_time = sim_settings.equilibration_length.to('femtosecond')
            equil_steps = round(equil_time / (ts * mc_steps))
            sampler.equilibrate(equil_steps.m)
            
            # production
            prod_time = sim_settings.production_length.to('femtosecond')
            prod_steps = round(prod_time / (ts * mc_steps))
            sampler.extend(prod_steps.m)
```

to-do there's an expectation here that production_time is a multiple of the integrator's n_steps. This should be checked explicitly & users warned.